emcc.py

```diff
@@ -2074,6 +2074,11 @@ def default_setting(name, new_default):
   else:
     settings.JS_LIBRARIES.append((0, 'library_pthread_stub.js'))
 
+  if settings.FORMAT_EXCEPTION_SUPPORT:
+    settings.DEFAULT_LIBRARY_FUNCS_TO_INCLUDE += ['$formatException']
+    settings.EXPORTED_FUNCTIONS += ["_emscripten_format_exception", "_free"]
+    settings.EXPORTED_RUNTIME_METHODS += ["formatException"]
```

Comment:
I guess if we want to avoid adding the setting, I could say something like:
```py
  if "formatException" in settings.EXPORTED_RUNTIME_METHODS:
    settings.DEFAULT_LIBRARY_FUNCS_TO_INCLUDE += ['$formatException']
    settings.EXPORTED_FUNCTIONS += ["_emscripten_format_exception", "_free"]
```